definitions.h

```diff
@@ -397,6 +448,101 @@ class VertexShader
         }
 };
 
```

Comment:
It would be better to overload the * operator here to be able to multiply them for you. I know it can be tricky to the overloaded operator to work.